flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamNode.java

```diff
@@ -209,6 +210,8 @@ public void setResources(ResourceSpec minResources, ResourceSpec preferredResour
     public void setManagedMemoryUseCaseWeights(
             Map<ManagedMemoryUseCase, Integer> operatorScopeUseCaseWeights,
             Set<ManagedMemoryUseCase> slotScopeUseCases) {
+        validateManagedMemoryUseCaseWeights(
+                managedMemoryOperatorScopeUseCaseWeights, operatorScopeUseCaseWeights);
```

Comment:
I think the requirement that no existing weights should be overwritten does not come from `StreamNode`, but from `TransformationTranslator`. Therefore, I'd suggest to enforce this validation in `SimpleTransformationTranslator` rather than here.